src/librustc_mir/interpret/operand.rs

```diff
@@ -353,6 +353,30 @@ impl<'mir, 'tcx, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
         Ok(str)
     }
 
+    // Turn the wide MPlace into a OsString (must already be dereferenced!)
+    pub fn read_os_str_from_wide_str(
+        &self,
+        mplace: MPlaceTy<'tcx, M::PointerTag>,
+    ) -> InterpResult<'tcx, std::ffi::OsString> {
+        let len = mplace.len(self)? as u64;
+        let mut u16_vec: Vec<u16> = Vec::with_capacity(len as usize);
+        for idx in 0..len {
+            let place = self.mplace_field(mplace, idx)?;
+            match self.read_scalar(OpTy::from(place))? {
+                ScalarMaybeUndef::Scalar(scalar) => {
+                    let item = Scalar::to_u16(scalar)?;
+                    u16_vec.push(item);
+                }
+                ScalarMaybeUndef::Undef => panic!("ScalarMaybeUndef: Undef error"),
+            }
+        }
+        let fetched_string = std::char::decode_utf16(u16_vec.into_iter())
```

Comment:
```suggestion
        let fetched_string = String::from_utf16(u16_vec);
```

Reply:
This is incorrect if the wide string has any lone surrogates in it as it would not be valid UTF-16. The correct thing to do is `OsStringExt::from_wide` although that gets tricky if this needs to run on non-Windows.

Reply:
So probably this is a reasonable fallback on non-Windows hosts? It will error if there is a surrogate, but work correctly otherwise?